### PR TITLE
Pool entity modifiers and remove unnecessary `Sprite` pooling

### LINKED ISSUE #347: Hit circle game objects are pooled too early
As the title says. This causes its animation to not finish properly after judgement, and its state will remain until it is reused, as shown in the video below.

https://github.com/osudroid/osu-droid/assets/52914632/2dd3ce8a-c6c1-4267-be05-028c9377cf1e

### PR DESCRIPTION
**Pull request by Reco1I**
* Fixes #347 

### Reason

Game objects are already pooled by `GameObjectPool` and its `Sprite` instances too. There's no reason to pool the sprites rather than worsen the performance.
In the other hand many entity modifiers instances are created per object spawn.

### CODE REVIEW

File: src/ru/nsu/ccfit/zuev/osu/game/Spinner.java [hunk 1]
```diff
@@ -290,8 +286,8 @@ public void update(final float dt) {
         if (percentfill > 1 || clear) {
             percentfill = 1;
             if (!clear) {
-                clearText = SpritePool.getInstance().getCenteredSprite(
-                        "spinner-clear", new PointF(center.x, center.y * 0.5f));
+                final PointF pos1 = new PointF(center.x, center.y * 0.5f);
+                clearText = new CentredSprite(pos1.x, pos1.y, ResourceManager.getInstance().getTexture("spinner-clear"));
```
Comment (Rian8337) [inline src/ru/nsu/ccfit/zuev/osu/game/Spinner.java:290]: Same as above

File: src/ru/nsu/ccfit/zuev/osu/game/Slider.java [hunk 2]
```diff
@@ -320,10 +322,8 @@ public void init(final GameObjectListener listener, final Scene scene,
         }
         ticks.clear();
         for (int i = 1; i <= tickCount; i++) {
-            final Sprite tick = SpritePool.getInstance().getCenteredSprite(
-                    "sliderscorepoint",
-                    getPercentPosition((float) (i * tickInterval
-                            / (maxTime * GameHelper.getTickRate())), null));
+            final PointF pos1 = getPercentPosition((float) (i * tickInterval / (maxTime * GameHelper.getTickRate())), null);
+            final Sprite tick = new CentredSprite(pos1.x, pos1.y, ResourceManager.getInstance().getTexture("sliderscorepoint"));
```
[the remark this answers is not in the document]
Comment (Reco1I) [reply src/ru/nsu/ccfit/zuev/osu/game/Slider.java:326]: Sure I'll pool them separately

File: src/ru/nsu/ccfit/zuev/osu/game/Slider.java [hunk 3]
```diff
@@ -732,15 +729,15 @@ public void update(final float dt) {
             approachCircle.clearEntityModifiers();
             approachCircle.setAlpha(0);
 
-            ball = SpritePool.getInstance().getAnimSprite("sliderb",
-                    SkinManager.getFrames("sliderb"));
+            int count = SkinManager.getFrames("sliderb");
+            ball = new AnimSprite(0, 0, "sliderb", count, count);
```
[the remark this answers is not in the document]
Comment (Reco1I) [reply src/ru/nsu/ccfit/zuev/osu/game/Slider.java]: Ball instantiation was moved to class constructor.

File: src/ru/nsu/ccfit/zuev/osu/game/Slider.java [hunk 4]
```diff
@@ -732,15 +729,15 @@ public void update(final float dt) {
...
             ball.setFps((float) (0.1f * GameHelper.getSpeed() * scale / timing.getBeatLength()));
             ball.setScale(scale);
             ball.setFlippedHorizontal(false);
 
             ball.registerEntityModifier(new FadeInModifier(0.1f * GameHelper.getTimeMultiplier()));
 
-            followCircle = SpritePool.getInstance().getSprite("sliderfollowcircle");
+            followCircle = new Sprite(0, 0, ResourceManager.getInstance().getTexture("sliderfollowcircle"));
```
[the remark this answers is not in the document]
Comment (Reco1I) [reply src/ru/nsu/ccfit/zuev/osu/game/Slider.java]: Follow circle instantiation was moved to class constructor.

File: src/ru/nsu/ccfit/zuev/osu/game/Spinner.java (hunk 1 above)
Comment (Reco1I) [reply src/ru/nsu/ccfit/zuev/osu/game/Spinner.java]: `clearText` instantiaton was moved to class constructor.

**Comment by Reco1I** [comment]
PR is ready

File: src/ru/nsu/ccfit/zuev/osu/game/FollowTrack.java [hunk 5]
```diff
@@ -55,21 +56,19 @@ public void init(final GameObjectListener listener, final Scene scene,
...
             if (frameCount == 1) {
-                point = SpritePool.getInstance().getCenteredSprite(
-                        "followpoint", pos);
+                point = new CentredSprite(x, y, ResourceManager.getInstance().getTexture("followpoint"));
             } else {
-                point = SpritePool.getInstance().getAnimSprite("followpoint-",
-                        frameCount);
-                point.setPosition(pos.x - pointSize * 0.5f, pos.y - pointSize
-                        * 0.5f);
+                point = new AnimSprite(0, 0, "followpoint-", frameCount, frameCount);
```
Comment (Rian8337) [inline src/ru/nsu/ccfit/zuev/osu/game/FollowTrack.java:70]: Ah, missed this. I believe this would also benefit from sprite pooling?

**Pull request merged**